## **Введение в программирование на python. Простейшие математические вычисления**
#### **Автор: Калитвин В.А.**
#### kalitvin@gmail.com

## **Арифметические операторы**

`+`  сложение (конкатенация)

In [ ]:
print(2 + 3)

5


`-`  вычитание

In [ ]:
print(8 - 3)

5


`*`  умножение (повторение)

In [ ]:
print(2 * 2)

4


`/`  деление

In [ ]:
print(8 / 4)

2.0


`//` деление с усечением дробной части

In [ ]:
print(8 // 3)

2


`**` возведение в степень

In [ ]:
print(2**3)

8


`%`  остаток от деления

In [ ]:
print(7 % 2)

1


Группировка в выражениях осуществляется с помощью круглых скобок.

In [ ]:
print(5 + 3 * 4) # сначала выполняется умножение, потом сложение

17


In [ ]:
print( (5 + 3) * 4) # сначала выполняется сложение, затем умножение

32


## Пример.
Вычислить $\frac{3+\frac{5}{3}}{\frac{3}{14}+(\frac{5}{14}+2)\cdot 3}$

In [ ]:
print((3 + 5 / 3) / (3 / 14 + (5 / 14 + 2) * 3 ))

0.6405228758169935


##***Модуль math***##

Модуль math предоставляет доступ для работы с математическими функциями.

Для подключения модуля необходимо в начало программы добавить строку

In [ ]:
import math

После этого можно вызывать функции из модуля math

In [ ]:
print(math.sin(0))
print(math.sin(math.pi/2))
print(math.factorial(3))

0.0
1.0
6


В вычислениях часто встречаются функции

pow(x,n) $x^n$

sqrt(x) $\sqrt{x}$

cbrt(x) $\root 3\of x$

fabs(x) $|x|$

exp(x) $e^x$

log(x) $ln x$

log10(x) $log_{10}(x)$

log2(x) $log_2(x)$

sin(x) $sin x$ (x в радианах)

cos(x) $cos x$ (x в радианах)

tan(x) $tan 𝑥$ (x в радианах)

asin(x) $arcsin x$ (в радианах)

acos(x) $arccos x$ (в радианах)

atan(x) $arctan x$ (в радианах)

ceil(x) Округление x сверху. Возвращает наименьшее целое значение, которое не меньше x

floor(x) Округление x снизу. Возвращает наибольшее целое значение, которое не больше x

round(x) Округление x до ближайшего целого (метод симметричного округления)

trunc(x) Округление x до ближайшего целого (метод отбрасывания дробной части)

fmod(a,b) Остаток от деления a на b




### **Преобразование углов** ###

math.degrees(x)

    Преобразует угол x из радиан в градусы.

math.radians(x)

    Преобразует угол x из градусов в радианы.



In [ ]:
print(math.degrees(math.pi))
print(math.radians(360))

180.0
6.283185307179586


### **Константы** ###

math.pi Число 𝜋 = 3.141592…

math.e  Число 𝑒 = 2.718281…

math.tau Константа τ = 6.283185….     
Константа $\tau$ равна отношению длины C окружности к её радиусу r: $\tau = \frac{C}{r}$. $\tau = 2\pi.$

math.inf Машинная бесконечность

In [ ]:
import math
print(math.pi)
print(math.e)
print(math.tau)
print(math.inf)

3.141592653589793
2.718281828459045
6.283185307179586
inf


In [ ]:
print(2.0**(-1075))

0.0


In [ ]:
print(2.0**1024)

OverflowError: (34, 'Numerical result out of range')

In [ ]:
print(1.0 + 2.0**(-53))

1.0


In [ ]:
a = 123456789123456789.0
b = 123456789123456788.0
f = a - b
print(f)

0.0


In [ ]:
if abs(0.1 + 0.2 - 0.3) < 0.0000000000000001:
  print('yes')
else:
  print('no')

yes


In [ ]:
if abs((0.1 + 0.2) - 0.3) < 0.0000000000001:
  print('yes')
else:
  print('no')

yes


In [ ]:
a = float(123456789123456789)
b = float(123456789123456788)
f = a - b
print(f)


0.0


In [ ]:
print(1 + 2**(-52))

1.0000000000000002


In [ ]:
print(2.0**1100)

OverflowError: (34, 'Numerical result out of range')

math.nan Not a number. Не число. В соответствии со стандартом IEEE-754, math.nan и float('nan') не равны любым числовым значениям, включая их самих. Для проверки, является ли объект объектом NaN, нужно использовать функцию isnan() вместо ==.

In [ ]:
import math
print(math.nan == math.nan) # False
print(float('nan') == float('nan')) #False
print(math.isnan(math.nan)) # True
print(math.isnan(float('nan'))) # True

False
False
True
True


In [ ]:
print(math.factorial(5))

120
